bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    harness::{Buffer, Mode, TlsImpl},
+    read_to_bytes, CA_CERT_PATH, SERVER_CERT_CHAIN_PATH, SERVER_KEY_PATH,
+};
+use log::info;
+use s2n_tls::{
+    callbacks::VerifyHostNameCallback,
+    config::{Builder, Config},
+    connection::Connection,
+    enums::Blinding,
+    security::DEFAULT_TLS13,
+};
+use std::{
+    ffi::c_void,
+    io::{Read, Write},
+    os::raw::c_int,
+    pin::Pin,
+    task::Poll::Ready,
+};
+
+pub struct S2nTls {
+    c_to_s_buf: Pin<Box<Buffer>>, // need Pin to make sure C pointer to Buffer doesn't move
+    s_to_c_buf: Pin<Box<Buffer>>, // Buffer used in custom IO as context
+    c_config: Config,
+    s_config: Config,
+    c_conn: Connection,
+    s_conn: Connection,
+    c_handshaked: bool,
+    s_handshaked: bool,
+}
+
+/// Custom callback for verifying hostnames, need it to use s2n-tls safely
+struct HostNameHandler<'a> {
+    hostname: &'a str,
```

Comment:
perhaps this variable could be `expected_server_name` to make line 38 a bit easier to parse? Also might be good to call out the limitation/why that's fine in the comment (pretty much a hard coded allowlist of a single name)